File: CS-Examples/07_Conversion/SelectedRangeToPDF.vb
Imports Spire.Xls

Namespace SelectedRangeToPDF

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ConversionSample1.xlsx")

            ' Add a new worksheet to the workbook
            workbook.Worksheets.Add("newsheet")

            ' Copy the range A9:E15 from the first worksheet to the second worksheet
            workbook.Worksheets(0).Range("A9:E15").Copy(workbook.Worksheets(1).Range("A9:E15"), False, True)

            ' Auto-fit columns in the range A9:E15 of the second worksheet
            workbook.Worksheets(1).Range("A9:E15").AutoFitColumns()

            ' Specify the output file name for saving the second worksheet as PDF
            Dim output As String = "SelectedRangeToPDF.pdf"

            ' Save the second worksheet to a PDF file
            workbook.Worksheets(1).SaveToPdf(output)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/07_Conversion/SelectedRangeToPDF.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace SelectedRangeToPDF
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ConversionSample1.xlsx");

            //Add a new sheet to workbook
            workbook.Worksheets.Add("newsheet");

            //Copy your area to new sheet.
            workbook.Worksheets[0].Range["A9:E15"].Copy(workbook.Worksheets[1].Range["A9:E15"], false, true);

            //Auto fit column width
            workbook.Worksheets[1].Range["A9:E15"].AutoFitColumns();

            //Save the document
            string output = "SelectedRangeToPDF.pdf";
            workbook.Worksheets[1].SaveToPdf(output);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}
